### ISSUE TITLE
Password with special characters not recognised

### ISSUE THREAD

**Issue by Kathara** [h3imdall/ftp-remote-edit#181]
Atom 1.25.0
ftp-remote-edit 0.12.11
Win10 64-bit

I've installed Atom today together with ftp-remote-edit and found an issue I didn't have on an other device.
If the password of the ftp-connection contains special characters (such as $, ~, etc) there's an error message that the password was "not understood".
If the password contains an @ it seems to be misinterpreted as address (

> getaddrinfo EAI_FAIL r5\EeboXfM9;~@2W.6BU:21

the last part being the password).

All ftp-servers I've added have a host, a username and a corresponding password. The ones that do not contain special characters have no problems at all.
In debug mode there's no error-message in the console.

As stated already, on the other device I've been connecting to these servers without problem. I believe I was missing out on the last update of ftp-remote-edit and had an older 1.2x version of Atom installed (can't verify).
I would love to use this package in the future too as it covers all the functionality I need in a very simple and "attractive" way but I would need this issue to be fixed or have a solution for it as I cannot change the passwords.
Or maybe, did I do something wrong? Or is there some sort of configuration that would fix the issue?

**Comment by FabrizioCaldarelli** [comment]
I tryed creating an ftp user with password $!@ and $$!!@@ and connection works fine.
Then I also tryed sftp connection with same passwords, but no errors.

**Comment by Kathara** [comment]
Thank you for your reply.

Strange. What could cause this problem then? Because I really only have this problem with the connections that have a password with special characters. And the error message comes from ftp-remote-edit. And with the other device and the same kind of connection information I didn't have this issue. Is there some kind of configuration that could cause this error?

What I have noted and may be related: after I've added the passwords with special characters in the servers-list, saved and reopened it, the password was way longer (a lot more dots than should be). Is there some kind of conversion happening (maybe with the special characters?) during the saving process? Is this a normal behaviour?

Could you test again with the password I have had the "getaddrinfo"-error above?
Thanks for your help

**Comment by FabrizioCaldarelli** [comment]
Sure to filling correct inputs? Because seems that you are insert username or password instead host.
What is username? 
What is password?
Is it a ftp connection (not sftp), is it right?
So I can replicate the connections with same parameters.

**Comment by Kathara** [comment]
Yes, I'm absolutely sure that I've entered the correct information at the correct place. And as I have stated before: on the other device I had the same server-setup and it worked.
It's a ftp-connection.
The host is something like: ftp.test.com
The username contains an @ just as other usernames of connections that work.
The password is r5\EeboXfM9;~@2W.6BU:21
It's only the password that is causing a problem. In both errors only the password is mentioned. One error states that the password is "not understood" the other is the error that I've quoted in my first post.
And this is what the server-mask looks like with the password above:

![image](https://user-images.githubusercontent.com/15015926/37704185-7d630a4c-2cf8-11e8-8949-a57ca448d0cd.png)

**Comment by FabrizioCaldarelli** [comment]
I've tried with these parameters a ftp connection (port 21):
username: prova@example.com
password: r5\EeboXfM9;~@2W.6BU:21

and it works. 

Has this ftp account same specific configuration?

**Comment by Kathara** [comment]
I think I'll just delete the server and try to add it again. I've tested this before and it hasn't worked. But maybe this time. I'll report back. Thanks a lot for your effort.

**Comment by FabrizioCaldarelli** [comment]
You are welcome

**Comment by cschindl** [comment]
Hi @Kathara,

the password is too long in the screenshot. It seems that several server entries have been merged here. Did other server entries disappear as a result?

**Comment by Kathara** [comment]
I have finally found the probable reason why it had this long password.
To answer your previous question: no, no other servers have disappeared.
I have deleted and recreated the server with the same name several times. Now that I have entered a different name, the connection information was correct and I could connect. Also the password has now the appropriate length in the field. 
So it seems as if when you delete a server, the data is not completely deleted and if you recreate a server with the same name as a deleted server the password you enter seems to be added to the password of the deleted server with the same name.
